google-cloud-clients/google-cloud-pubsub/src/main/java/com/google/cloud/pubsub/v1/Publisher.java

```diff
@@ -351,21 +384,45 @@ public void onFailure(Throwable t) {
               messagesWaiter.incrementPendingMessages(-outstandingBatch.size());
             }
           }
-        },
-        directExecutor());
+        };
+
+    if (outstandingBatch.orderingKey == null || outstandingBatch.orderingKey.isEmpty()) {
```

Comment:
Consider an encapsulating method for `hasOrderingKey()`

Reply:
Great idea.